### ISSUE TITLE
Should a rewrite unlock a data source from exhaustion? 

### ISSUE THREAD

**Issue by inf3rno** [inf3rno/dataflower#108]
#107 With the new design I was wondering whether it is really necessary to exhaust a source for eternity. We could add different states, e.g. paused, stopped, etc... to these sources. I need examples which justify these modifications. Probably reading multiple files, arrays, etc. as a single source. So after file ends, the source will be exhausted and at the beginning of every file, it will be opened again or something like that. I am not sure whether this should be a general feature, or just let people implement it by their sources on their own. Maybe I could add a lock/unlock mechanism, I can call that open/close. So opening and closing will be temporary while exhaustion will be permanent.

**Comment by inf3rno** [comment]
Certainly not. The block is there because the flow is no more sustainable. Maybe later I'll add an unblock method, but certainly not now.
